### PR TITLE
add 3dt checks runner

### PR DESCRIPTION
**Pull request by darkonie**
## High Level Description

[EE DC/OS PR](https://github.com/mesosphere/dcos-enterprise/pull/1025)

Refactor 3dt to support subcommands. Add dcos-checks runner as `3dt check ...` subcommand.

With this PR 3dt has been refactored to use subcommands:
  - `3dt daemon` will start the usual server.
  - `3dt check ...` invoke dcos-checks.

Bump `dcos-vagrant`
3dt does not support `-diag` flag anymore. A [PR](https://github.com/dcos/dcos-vagrant/pull/101) to fix this was merged.

## Related Issues

  - [DCOS-14883](https://jira.mesosphere.com/browse/DCOS-14883)
  - [DCOS-15801](https://jira.mesosphere.com/browse/DCOS-15801)

## Checklist for all PR's

  - [X] Included a test which will fail if code is reverted but test is not. If there is no test please explain here:
  - [X] Read the [DC/OS contributing guidelines](https://github.com/dcos/dcos/blob/master/contributing.md)
  - [X] Followed relevant code rules [Rules for Packages and Systemd](https://github.com/dcos/dcos/tree/master/docs)

## Checklist for component/package updates:

If you are changing components or packages in DC/OS (e.g. you are bumping the sha or ref of anything underneath `packages`), then in addition to the above please also include:

  - [X] Change log from the last version integrated (this should be a link to commits for easy verification and review):[dcos-3dt](https://github.com/dcos/3dt/compare/206ba11a3ec02d02f95bab6aa3cec39cd7395bb3...6b82d3c84f2a72e0ee44cfaf62294dbe79e1b69a)
  - [X] Test Results: [dcos-3dt](https://jenkins.mesosphere.com/service/jenkins/job/public-dcos-3dt-pulls/371/console)
  - [X] Code Coverage (if available): [dcos-3dt](https://jenkins.mesosphere.com/service/jenkins/job/public-dcos-3dt-pulls/371/console)
___
**PLEASE FILL IN THE TEMPLATE ABOVE** / **DO NOT REMOVE ANY SECTIONS ABOVE THIS LINE**

## Instructions and Review Process

**What is the review process and when will my changes land?**

All PRs require 2 approvals using github's [pull request reviews](https://help.github.com/articles/about-pull-request-reviews/).

Reviewers should be:
* Developers who understand the code being modified.
* Developers responsible for code that interacts with or depends on the code being modified.

It is best to proactively ask for 2 reviews by @mentioning the candidate reviewers in the PR comments area. The responsibility is on the developer submitting the PR to follow-up with reviewers and make sure a PR is reviewed in a timely manner. Once a PR has **2 ship-it's**, **no red reviews**, and **all tests are green** it will be included in the [next train](https://github.com/dcos/dcos/blob/master/contributing.md).

### CODE REVIEW

**Comment by darkonie** [comment]
@mesosphere-mergebot bump-ee

**Comment by darkonie** [comment]
@mesosphere-mergebot bump-ee

**Comment by darkonie** [comment]
@mesosphere-mergebot bump-ee

**Comment by darkonie** [comment]
@mesosphere-mergebot bump-ee

File: gen/dcos-config.yaml [hunk 1]
```diff
@@ -512,6 +512,101 @@ package:
...
+              "timeout": "1s"
+            },
+            "check_mesos_metrics_master": {
+              "description": "check if mesos replog is synchronized",
+              "cmd": ["/opt/mesosphere/bin/dcos-checks", "--role", "master", "mesos-metrics"],
+              "timeout": "1s",
+              "roles": ["master"]
+            },
+            "check_mesos_metrics_agent": {
+              "description": "check if mesos replog is synchronized",
```
Comment (branden) [inline gen/dcos-config.yaml]: Mesos agents don't have a replog.

File: gen/dcos-config.yaml [hunk 2]
```diff
@@ -512,6 +512,101 @@ package:
...
+              "cmd": ["/opt/mesosphere/bin/dcos-checks", "executable", "unzip"],
+              "timeout": "1s"
+            },
+            "check_ip": {
+              "description": "validate detect_ip script output",
+              "cmd": ["/opt/mesosphere/bin/dcos-checks", "ip"],
+              "timeout": "1s"
+            },
+            "check_mesos_metrics_master": {
+              "description": "check if mesos replog is synchronized",
```
Comment (branden) [inline gen/dcos-config.yaml]: Should the check name mention replog, not metrics? Looks like the Mesos metrics API is just the means to the end of knowing whether the replog is synced.

File: gen/dcos-config.yaml (hunk 1 above)
Comment (darkonie) [inline gen/dcos-config.yaml]: @amitaekbote do we need to run this check on agents?

File: gen/dcos-config.yaml (hunk 2 above)
Comment (darkonie) [inline gen/dcos-config.yaml]: @amitaekbote would you suggest an appropriate name?

Comment (amitaekbote) [inline gen/dcos-config.yaml]: The description is fine, we can change the name of the file as a separate commit, @branden you ok with that?

File: gen/dcos-config.yaml (hunk 1 above)
Comment (amitaekbote) [inline gen/dcos-config.yaml]: we check the same endpoint to verify if an agent has rejoined the cluster

Comment (darkonie) [inline gen/dcos-config.yaml]: does it make sense to run on masters only?

File: gen/dcos-config.yaml (hunk 2 above)
Comment (branden) [inline gen/dcos-config.yaml]: I was referring to the name of the check in config, not the executable it calls. That part I'm not too worried about. But the check name will show up in runner output, so it's worth making sure that piece is clear and accurate.

File: gen/dcos-config.yaml (hunk 1 above)
Comment (branden) [inline gen/dcos-config.yaml]: We'll need this on agents. It just shouldn't mention a replog, since agents don't have one.

Comment (amitaekbote) [inline gen/dcos-config.yaml]: Nope, we need this. Today we run this after upgrade test to verify that an agent node has upgraded successfully

File: gen/dcos-config.yaml (hunk 2 above)
Comment (amitaekbote) [inline gen/dcos-config.yaml]: @darkonie  check_mesos_replog_synchronized?

File: test_util/run-all [hunk 3]
```diff
@@ -33,7 +33,7 @@ else
 fi
 
 # Pin to just the right dcos-vagrant commit
-git -C dcos-vagrant checkout -qf 2b323c4c1942e3f6d484a7375e749929356a4417
+git -C dcos-vagrant checkout -qf 03de5522ba6da57b3f624fcd9985545f70a89497
```
Comment (branden) [inline test_util/run-all]: It's unclear why this change and the one below are necessary. Can you add something to the PR description that explains this?

File: gen/dcos-config.yaml [hunk 4]
```diff
@@ -519,6 +519,97 @@ package:
...
+              "cmd": ["/opt/mesosphere/bin/dcos-checks", "ip"],
+              "timeout": "1s"
+            },
+            "check_mesos_replog_synchronized_master": {
+              "description": "check if mesos replog is synchronized",
+              "cmd": ["/opt/mesosphere/bin/dcos-checks", "--role", "master", "mesos-metrics"],
+              "timeout": "1s",
+              "roles": ["master"]
+            },
+            "chech_agents_rejoined_cluster": {
```
Comment (branden) [inline gen/dcos-config.yaml]: `check_agent_registered_with_masters`? `rejoined` is presumptive -- rejoined after what?

File: gen/dcos-config.yaml [hunk 5]
```diff
@@ -519,6 +519,97 @@ package:
...
+              "description": "check if mesos replog is synchronized",
```
Comment (branden) [inline gen/dcos-config.yaml]: This description shouldn't mention replog, since agents don't have one.

File: test_util/run-all (hunk 3 above)
Comment (darkonie) [inline test_util/run-all]: sure, the main reason we had to add this commit is because `-diag` is not supported by 3dt anymore. Latest dcos-vagrant bump fixes it.

File: gen/dcos-config.yaml (hunk 5 above)
Comment (darkonie) [inline gen/dcos-config.yaml]: if you think the description is not appropriate here, please suggest one.

Comment (darkonie) [inline gen/dcos-config.yaml]: @amitaekbote

Comment (branden) [inline gen/dcos-config.yaml]: name: `check_mesos_agent_registered_with_masters`
description: `Check if the Mesos agent has registered with the masters.`

**Comment by amitaekbote** [comment]
@mesosphere-mergebot label ship-it

**Comment by branden** [comment]
@mesosphere-mergebot label work in progress

**Comment by amitaekbote** [comment]
@mesosphere-mergebot label ship it
